independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResponseHandler.java

```diff
@@ -99,9 +99,6 @@ public void handle(ResteasyReactiveRequestContext requestContext) throws Excepti
                 responseBuilder = fromResponse(existing);
                 responseBuilder.entity(genericEntity.getEntity());
             } else {
-                // TCK says to use the entity type as generic type if we return a response
-                if (existing.hasEntity() && (existing.getEntity() != null))
-                    requestContext.setGenericReturnType(existing.getEntity().getClass());
```

Comment:
This is only needed for `Response` (for which we still have it), but `RestResponse` is has a generic type, so this is not needed